syntax="proto3";

option go_package = "greet/v1/go-greet";

package greet.v1;

service Greeter {
    rpc SayHello (HelloRequest) returns (HelloReply) {}
    rpc SecureSayHello (HelloRequest) returns (HelloReply) {}
    rpc TransactionSayHello (HelloRequest) returns (HelloReply) {}
    rpc PersistedSayHello (HelloRequest) returns (HelloReply) {}
}

message HelloRequest {
    string name = 1;
}
  
message HelloReply {
    string message = 1;
}